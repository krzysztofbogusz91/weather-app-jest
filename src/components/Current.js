import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchWeather, fetchUser, fetchForecast } from '../actions/index';
import { Table } from '../containers/Table';


export class Current extends Component {
    constructor(props) {
        super(props);
        this.state = {
            search: "",
            showTable: false
        }

    }

    setAll = async () => {
        await this.props.fetchUser()
        const cord = this.props.cords;
        await this.props.fetchForecast(cord)
        await this.props.fetchWeather(cord);
        this.setState({
            showTable: this.props.weather.length > 0 ? true : false
        })

    }


    componentDidMount() {
        this.setAll()
      
        //RELOAD EVERY 10 minutes? and store it all in Local storage?
        // this.time = setTimeout(()=>{
        //     this.setAll();
        //     console.log('reload'); 
        // },2000)
    }
    //NOT SOLVING PROBLEM
    // componentWillUnmount(){
    //     this.setAll().abort()
    // }

    inputChange = (e) => {
        this.setState({
            search: e.target.value
        })

    }

    tables() {
        //reduce to avoid using [0]
       const weather = this.props.today.weather.reduce((acc,bcc)=>acc.concat(bcc))
       weather.name = this.props.today.name;
       weather.temp = this.props.today.main.temp
       weather.humidity = this.props.today.main.humidity 
       
       const forecast = this.props.weather.map(a =>{
        const weat = a.weather[0];
        weat.name = a.dt_txt;
        weat.temp = a.main.temp
        weat.humidity = a.main.humidity
        return weat 
       })
       
        return (
            <div>
                <Table type="today" table={[weather]} />
                <Table type="forecast" table={forecast}/>
            </div>
        )
    }

    render() {
        //console.log(this.props)
        //console.log(this.state.showTable)


        return (
            <div className="mt-5">
                <form action="submit ">
                    <input
                        value={this.state.search}
                        onChange={this.inputChange}
                        type="text"
                        className='search-fetch form-control' />
                </form>
                <div className='current-weather'>
                    {this.state.showTable ? this.tables() : null}
                </div>
            </div>
        );
    }
}

Current.propTypes = {
    fetchWeather: PropTypes.func.isRequired,
    fetchUser: PropTypes.func.isRequired,
    fetchForecast: PropTypes.func.isRequired,
    today: PropTypes.any,
    weather: PropTypes.any,
    cords: PropTypes.any,
};

Current.defaultProps = {
    today: [],
    weather: [],
    cords: {},
}

export default connect(state => ({ ...state }), { fetchWeather, fetchUser, fetchForecast })(Current);
