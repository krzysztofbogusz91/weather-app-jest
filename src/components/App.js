import React, { Component } from 'react';
import Current from "./Current";
import { Route, BrowserRouter } from 'react-router-dom';
import { Header } from '../containers/Header';
import User from './User'
import Search from './Search';
import Contact from './Contact';

// TODO 
//-- Add default route and 404 page
class App extends Component {
    render() {
        return (
        <div className={'main container'} >
            <BrowserRouter>
                <div>
                    <Header />
                    <Route exact path='/' component={ Current }/>
                    <Route path='/search' component={ Search }  />
                    <Route path='/user' component={ User }  />
                    <Route path='/contact' component={ Contact }  />
                </div>
            </BrowserRouter>
        </div>
    );
  }
}


export default App;
