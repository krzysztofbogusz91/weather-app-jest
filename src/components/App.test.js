import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import Current from './Current'
import { shallow, mount } from 'enzyme';
import set from '../enzyme_set_up';

describe('App',()=>{
    const app = shallow( <App />)

    it('renders without crashing', () => {
        expect(app).toMatchSnapshot();
    });
      
})


