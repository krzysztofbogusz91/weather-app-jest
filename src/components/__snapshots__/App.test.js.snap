// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <BrowserRouter>
        <div>
                <Unknown />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        path="/search"
                />
                <Route
                        component={[Function]}
                        path="/user"
                />
                <Route
                        component={[Function]}
                        path="/contact"
                />
        </div>
</BrowserRouter>,
      "className": "main container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Unknown />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    path="/search"
          />
          <Route
                    component={[Function]}
                    path="/user"
          />
          <Route
                    component={[Function]}
                    path="/contact"
          />
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              path="/search"
/>,
            <Route
              component={[Function]}
              path="/user"
/>,
            <Route
              component={[Function]}
              path="/contact"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/search",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/user",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/contact",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <BrowserRouter>
          <div>
                    <Unknown />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              path="/search"
                    />
                    <Route
                              component={[Function]}
                              path="/user"
                    />
                    <Route
                              component={[Function]}
                              path="/contact"
                    />
          </div>
</BrowserRouter>,
        "className": "main container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <Unknown />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        path="/search"
            />
            <Route
                        component={[Function]}
                        path="/user"
            />
            <Route
                        component={[Function]}
                        path="/contact"
            />
</div>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown />,
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                path="/search"
/>,
              <Route
                component={[Function]}
                path="/user"
/>,
              <Route
                component={[Function]}
                path="/contact"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/search",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/user",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/contact",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
